CSharp/Library/Microsoft.Bot.Builder/Dialogs/IDialogContext.cs

```diff
@@ -1,91 +1,91 @@
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED ""AS IS"", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
 using System.Collections.Generic;
-using System.Threading;
-using System.Threading.Tasks;
-
-using Microsoft.Bot.Builder.Dialogs.Internals;
-using Microsoft.Bot.Connector;
-
-namespace Microsoft.Bot.Builder.Dialogs
-{
-    /// <summary>
-    /// Encapsulates a method that represents the code to execute after a result is available.
-    /// </summary>
-    /// <remarks>
-    /// The result is often a message from the user.
-    /// </remarks>
-    /// <typeparam name="T">The type of the result.</typeparam>
-    /// <param name="context">The dialog context.</param>
-    /// <param name="result">The result.</param>
-    /// <returns>A task that represents the code that will resume after the result is available.</returns>
-    public delegate Task ResumeAfter<in T>(IDialogContext context, IAwaitable<T> result);
-
-    /// <summary>
-    /// Encapsulate a method that represents the code to start a dialog.
-    /// </summary>
-    /// <param name="context">The dialog context.</param>
-    /// <returns>A task that represents the start code for a dialog.</returns>
-    public delegate Task StartAsync(IDialogContext context);
-
-    /// <summary>
-    /// The context for the bot.
-    /// </summary>
-    public interface IBotContext : IBotData, IBotToUser
-    {
-        CancellationToken CancellationToken { get; }
-
-        /// <summary>
-        /// The activity posted to bot.
-        /// </summary>
-        /// <remarks> This is the incoming activity in reactive cases.
-        /// for proactive case, i.e. Conversation.ResumeAsync code path,
-        /// it will be the <see cref="IMessageActivity"/> returned by <see cref="ConversationReference.GetPostToBotMessage"/>.
-        /// </remarks>
-        IActivity Activity { get; }
-    }
-
-    /// <summary>
-    /// The context for the execution of a dialog's conversational process.
-    /// </summary>
-    public interface IDialogContext : IDialogStack, IBotContext
-    {
-    }
-
+using System;
+using Microsoft.Bot.Builder.Dialogs.Internals;

```

Comment:
Sort usings and we recently started using the following VS rule when sort usings: Place 'System' directives first when sorting usings